### PR TITLE
Convert controller files to structured logs

### PR DESCRIPTION
**Pull request by scottilee**
#### What type of PR is this?
/kind cleanup
/sig instrumentation

#### What this PR does / why we need it:
Converts log statements to use structured logging.

#### Which issue(s) this PR fixes:
N/A

#### Special notes for your reviewer:
N/A

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/1602-structured-logging/kep.yaml
```

### CODE REVIEW

**Comment by ehashman** [comment]
For any changes, add the directory to `hack/.structured_logging`, e.g. in this case, `pkg/controller/subdir/...` This will ensure that the file is linted to catch any missing migrations.

/ok-to-test
/remove-sig instrumentation
/area logging

**Comment by pacoxu** [comment]
`hack/.structured_logging` is not in alphabetical order. Please sort it.

**Comment by scottilee** [comment]
> `hack/.structured_logging` is not in alphabetical order. Please sort it.

Fixed.

**Comment by scottilee** [comment]
Addrssing "pull-kubernetes-verify" errors by removing additions to .structured_logging, since it includes all subdirectories.

**Comment by cheftako** [comment]
> For any changes, add the directory to `hack/.structured_logging`, e.g. in this case, `pkg/controller/subdir/...` This will ensure that the file is linted to catch any missing migrations.
> 
> /ok-to-test
> /remove-sig instrumentation
> /area logging

+1 to adding the suggested directory to https://github.com/kubernetes/kubernetes/blob/master/hack/.structured_logging

File: pkg/controller/controller_utils.go [hunk 1]
```diff
@@ -585,10 +585,10 @@ func (r RealPodControl) createPods(nodeName, namespace string, template *v1.PodT
 	}
 	accessor, err := meta.Accessor(object)
 	if err != nil {
-		klog.Errorf("parentObject does not have ObjectMeta, %v", err)
+		klog.ErrorS(err, "parentObject does not have ObjectMeta")
```
Comment (yangjunmyfm192085) [inline pkg/controller/controller_utils.go:588]:
```suggestion
		klog.ErrorS(err, "ParentObject does not have ObjectMeta")
```

File: pkg/controller/controller_utils.go [hunk 2]
```diff
@@ -602,7 +602,7 @@ func (r RealPodControl) DeletePod(namespace string, podID string, object runtime
 	klog.V(2).InfoS("Deleting pod", "controller", accessor.GetName(), "pod", klog.KRef(namespace, podID))
 	if err := r.KubeClient.CoreV1().Pods(namespace).Delete(context.TODO(), podID, metav1.DeleteOptions{}); err != nil {
 		if apierrors.IsNotFound(err) {
-			klog.V(4).Infof("pod %v/%v has already been deleted.", namespace, podID)
+			klog.V(4).InfoS("Pod has already been deleted.", "podNamespace", namespace, "pod", podID)
```
Comment (yangjunmyfm192085) [inline pkg/controller/controller_utils.go:605]: klog.KRef or klog.KObj？

File: pkg/controller/controller_utils.go (hunk 1 above)
Comment (cheftako) [reply pkg/controller/controller_utils.go:588]: Please do not capitalize error messages. We try to adhere to https://github.com/golang/go/wiki/CodeReviewComments#error-strings

**Comment by scottilee** [comment]
> Just realized you had this but removed it again. What was the issue which caused you to remove it?

It includes all sub-directories. Probably makes more sense to add it later when all sub-directories have been moved to structured logging.

File: pkg/controller/controller_utils.go (hunk 1 above)
Comment (scottilee) [reply pkg/controller/controller_utils.go:588]: Fixed.

File: pkg/controller/controller_utils.go (hunk 2 above)
Comment (scottilee) [reply pkg/controller/controller_utils.go:605]: klog.kObj wouldn't work and I need the pod name for klog.KRef, which podID sounds different from?

Comment (yangjunmyfm192085) [reply pkg/controller/controller_utils.go:605]: Is it okay to keep the format of L602?
Is it works?

**Comment by scottilee** [comment]
Rebased.

**Review by yangjunmyfm192085 — COMMENTED** [review]
/retest

**Pull request closed without merging**